index.bs

```diff
@@ -612,6 +652,105 @@ A mapping Web application uses the device's orientation to correctly align the m
+    1. If <var>beta</var> is not a {{Number}}, or its value is <strong>NaN</strong>, +∞, or −∞, return <strong>undefined</strong>.
+    1. If <var>beta</var> is not in the range [-180, 180), then return <strong>undefined</strong>.
+    1. Let <var>gamma</var> be the result of invoking <a>get a property</a> from <var>parameters</var> with "<code>gamma</code>".
+    1. If <var>gamma</var> is not a {{Number}}, or its value is <strong>NaN</strong>, +∞, or −∞, return <strong>undefined</strong>.
+    1. If <var>gamma</var> is not in the range [-90, 90), then return <strong>undefined</strong>.
+    1. Let <var>reading</var> be a new <a>map</a>.
+    1. <a for=map>Set</a> <var>reading</var>["<code>alpha</code>"] to <var>alpha</var>.
+    1. <a for=map>Set</a> <var>reading</var>["<code>beta</code>"] to <var>beta</var>.
+    1. <a for=map>Set</a> <var>reading</var>["<code>gamma</code>"] to <var>gamma</var>.
+    1. Return <var>reading</var>.
+
+Note: <var>reading</var> is defined as a <a>map</a> in the algorithm above to prevent a dependency on the <a>sensor reading</a> concept from the [[GENERIC-SENSOR]] specification. They should be interchangeable for the purposes of the algorithm above.
+
+</div>
+
+<h4 id="absolute-orientation-euler">The "absolute-orientation-euler" virtual sensor type</h4>
+
+The <a>per-type virtual sensor metadata</a> <a>map</a> must have the following <a for=map>entry</a>
+:
+: <a for=map>key</a>
+:: "<code>absolute-orientation-euler</code>"
+: <a for=map>value</a>
+:: A <a>virtual sensor metadata</a> whose <a>reading parsing algorithm</a> is <a>parse orientation data reading</a>.
+
+<h4 id="relative-orientation-euler">The "relative-orientation-euler" virtual sensor type</h4>
+
+The <a>per-type virtual sensor metadata</a> <a>map</a> must have the following <a for=map>entry</a>
+:
+: <a for=map>key</a>
+:: "<code>relative-orientation-euler</code>"
+: <a for=map>value</a>
+:: A <a>virtual sensor metadata</a> whose <a>reading parsing algorithm</a> is <a>parse orientation data reading</a>.
+
+Device Motion Automation {#device-motion-automation}
+------------------------
+
+The motion data retrieved from the platform by the user agent comes from accelerometers and gyroscopes. Implementations that also support the [[ACCELEROMETER]] and [[GYROSCOPE]] specifications can use the infrastructure defined in those specifications. This specification provides the same infrastructure for implementations that do not support said specifications.
+
+<h4 id="accelerometer-virtual-sensors">Accelerometer virtual sensors</h4>
+
+Accelerometer virtual sensors are used to provide <a>acceleration with gravity</a> data to the platform.
+
+If <a>per-type virtual sensor metadata</a>["<code>accelerometer</code>"] does not <a for=map>exist</a> (and is therefore not set by another specification), implementations must add the following <a for=map>entry</a> to <a>per-type virtual sensor metadata</a>:
```

Comment:
Why not move this section entirely out of [ACCELEROMETER] and the same below?

Reply:
Done.